### ISSUE TITLE
Per-weapon potion/armor damage modifiers

### ISSUE THREAD

**Issue by galacticwarrior9** [WeaponMechanics/MechanicsMain#220]
### Description

It is possible to modify how potion effects affect damage globally via `config.yml`:

```yaml
Damage:
  Minimum_Rate: 0.15
  Maximum_Rate: 4.0
  Potions:
    DAMAGE_RESISTANCE: -0.05
    UNLUCK: 0.15
    WEAKNESS: 0.05
```

It would be helpful if this could be modified on a per-weapon basis too like in CSP.

### Configuration

I imagine it looking the same in weapon files; it would just go under the damage section. For example:

```yaml
Damage:
  Base_Damage: 30
  Armor_Damage: 1
  Dropoff:
    - "50~-10"
  Potions:
    DAMAGE_RESISTANCE: -0.05
    UNLUCK: 0.15
    WEAKNESS: 0.05
```

### Payment (optional)

- [X] I'm willing to pay for this feature

**Comment by CJCrafter** [comment]
So I have wanted to recode the entire damage system. I'll make this offer and see if it appeals to you:

### The New System
Add a `Damage_Multipliers` section which contains the following options:

```yaml
Damage_Multipliers:
  Min: 0%  # Make sure the damage is never negative
  Max: 1000%  # Make sure the gun never does 10x damage
  Entities:
    - ZOMBIE 100%  # +100% damage to zombies
  Potions:
    - WEAKNESS 100%  # +100% damage to entities with weakness
  Armor_Multipliers:
    Only_Specific_Armor: <true/false>  # only use armor based on the damagepoint hit head/chest/legs/feet
    Types:  # I suggest not using this... use 'Per_Level' since it works better and is easier to use
      - leather 0%
      - gold 0%  # CJ REMEMBER GOLDEN FOR 1.12
      - diamond 0%
      - turtle 0%
    Per_Level: 0%  # Checks the player's entity armor attribute + the item's armor attribute (or all armor if Only_Specific_Armor: false)
    Enchantments:
      - protection 0%
      - projectile_projection 0%
  Movement:
    Sneaking: 0%
    Walking: 0%
    Swimming: 0%
    Sprinting: 0%
    In_Midair: 0%

```

There would be one of these in the `config.yml` file to be used for each weapon. **BUT**, if you wanted to change this per weapon, you could copy and paste all options and completely override the system per gun.  This is a complete recode of the damage feature and would also include some minor performance boosts to that system. I am willing to get all of this done for 70$ (Yeah, probably not what you were looking for. BUT, I bet you could find someone on discord willing to split the difference with you. Let me know.)


Fixes #139

**Comment by CJCrafter** [comment]
Implemented... working on the wiki for this now
